main.nf

```diff
@@ -92,9 +92,9 @@ Set species = ['hsa', 'eco', 'mmu', 'mau']
 if ( params.profile ) { exit 1, "--profile is WRONG use -profile" }
 if ( params.reads == '' ) { exit 1, "--reads is a required parameter" }
 if ( params.include_species && ! params.species ) { exit 1, "You need to select a species with --species " + species + " for automatic download." }
-if ( params.include_species == '' && params.genome == '' ) { exit 1, "You need to set a genome for mapping and a annotation for counting: with --include_species --species " + species + " are provided and automatically downloaded; with --genome and --annotation set csv files for custom input." }
+if ( (! params.include_species) && params.genome == '' && ! workflow.profile.contains('test') ) { exit 1, "You need to set a genome for mapping and a annotation for counting: with --include_species --species " + species + " are provided and automatically downloaded; with --genome and --annotation set csv files for custom input." }
```

Comment:
see my question on basecamp. If I try

```
nextflow run main.nf --reads eco.csv --species eco --cores 8 --assembly -profile local,docker
```

I get 

> You need to set a genome for mapping and an annotation for counting: with --include_species --species [hsa, eco, mmu, mau] are provided and automatically downloaded; with --genome and --annotation set csv files for custom input.

because `--include_species` is not set.

Reply:
yeah, right now, you need `--include-species` OR (`--genome` AND `--annotation`)

so this should work:
```
nextflow run main.nf --reads eco.csv --species eco --include_species --cores 8 --assembly -profile local,docker
```

Reply:
Okay, I think you explained that earlier... but why can we not just have 

* `--species eco`
* `--genome genome.csv --annotation anno.csv`
* `--species eco --genome genome.csv --annotation .csv`

So why do we need the `--include_species` parameter? Could `params.include_species` not just be set to `true` if `--species` is used? 

And: if it's difficult without, we should state clearly in the README that `--species` __needs__ `--include_species` as a parameter and a more specific help message if it is not provided.

Reply:
`--species xxx` provides also the info for downstream analysis in the DESeq2 part.

Good idea, to set `include_species` to `ture` if `--species` is used. But imagine for your last example: you don't wan to autodownload `eco` (because you already have it on your machine) AND you want downstream analysis, so you need `--species`.

We might rename the params. Do you think, that'll help?

Reply:
Ah okay... so `--species` is used as a label for two things

1) autodownload of genomes/annotations
2) downstream (pathway) analyses that can be performed

If I have a FASTA and GTF for _e. coli_ on my machine that I want to use and only provide `--genome --annotation` the downstream part will not run because I did not specify `--species`? So I need to add `--species eco` but I don't want to download the genome/anno data thus I don't use `--include_species`. And then there is of course the neat case where I provide a viral genome and annotation via `--genome --annotation` and can then define `--species hsa` to get a concatenation of virus and host. 

So all is great, but I think it's complicated and you are right and maybe some renaming of the variables might already help to solve potential confusion. Just for discussion (maybe there are better labels)

* `--genome` and `--annotation` are clear (except people might try at first to directly provide a FASTA and GTF, but which we can also allow in general?)
* `--species` could be split up into
    * `--species` or `--autodownload` to obtain FASTA/GTF automatically
    * `--pathway` or `--downstream` where we support a set of labels (`eco, hsa, mmu`, ...) for downstream analysis? 

The set of labels that we support for `--autodownload` and `--downstream` might be different. E.g. we allow autodownload of _Mesocricetus auratus_ but do not support downstream analysis for that species. 

Thus, we can then do 

* `--genome --annotation` w/o downstream analysis
* `--genome --annotation --pathway` w downstream analysis
* `--autodownload` w/o downstream analysis
* `--autodownload --pathway` w downstream analysis
* `--genome --annotation --autodownload` to concatenate local and downloaded FASTA/GTFs w/o downstream analysis
* `--genome --annotation --autodownload --pathway` to concatenate local and downloaded FASTA/GTFs w/ downstream analysis

does this cover all cases?

Reply:
yeah, I think, that's it!

This would introduces backwards incompatible changes, so I suggest:
- release 1.1.0 with the current state
- make the params changes and set `--species` and `--inlcude-species` deprecated -> minor release
- remove deprecated stuff with next major release

Reply:
yes, good plan!